docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,171 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Configuring Keycloak"
+summary="An overview of the server configuration">
+
+In this guide, you will learn the core concepts around the server configuration and how to configure the server using the different
+configuration options available. You will also learn how to properly configure the server to achieve an optimal runtime for faster
+startup and low memory footprint.
+
+== Understanding Server Options
+
+All the server options can be found by using the `help` option for individual commands:
+
+<@kc.all parameters="--help"/>
+
+You can instead look at all the server options at <@links.server id="all-config"/>.
+
+Server options are loaded from different sources in a specific order and they use different formats:
+
+|===
+|*Source* | *Format*
+
+|CLI
+|--<category>-<sub-category>-<property>=<value>
+
+|Environment Variable
+|KC_CATEGORY_SUB_CATEGORY_PROPERTY
+
+|`conf/keycloak.conf`
+|<category>-<sub-category>-<property>=<value>
+|===
+
+Given the `db-url-host` option as an example, you would set this property as follows:
+
+.Command-line argument
+<@kc.all parameters="--db-url-host=mykeycloakdb"/>
+
+.Environment variable
+```
+export KC_DB_URL_HOST=mykeycloakdb
+```
+
+.`conf/keycloak.conf`
+```
+db-url-host=mykeycloakdb
+```
+
+The configuration source and the corresponding format you should use is use-case specific. That decision depends on the platform the server is deployed to
+as well as on the optimizations you can get for an optimal runtime.
+
+== Configuration Stages
+
+The server options are narrowed to a specific command or configuration stage. This clear separation is one of the key aspects
+of the server configuration and is related to a series of optimizations that are done in order to deliver the best runtime when starting and running the server.
+
+Configuration is basically done in two stages:
+
+* Before starting the server in order to build an optimized server image for an optimal runtime
+* When starting the server
+
+The first stage involves running the `build` command using any **build option** available from this command. On the other hand, the second stage involves
+starting the server using any **configuration option** available from the `start` command.
+
+== Configuring the server
+
+The clear separation between build options and configuration options is one of the key aspects
```

Comment:
Is this sentence repeated on purpose? It also begins the "Configuration Stages" section.

Reply:
It was, but we can remove it.